stdlib/public/stubs/LibcShims.cpp

```diff
@@ -329,3 +330,52 @@ swift::_stdlib_cxx11_mt19937_uniform(__swift_uint32_t upper_bound) {
   std::uniform_int_distribution<__swift_uint32_t> RandomUniform(0, upper_bound);
   return RandomUniform(getGlobalMT19937());
 }
+
+#if defined(__APPLE__)
+#include <Security/Security.h>
+SWIFT_RUNTIME_STDLIB_INTERFACE
+void swift::_stdlib_random(void *buf, __swift_size_t nbytes) {
+  if (__builtin_available(macOS 10.12, iOS 10, tvOS 10, watchOS 3, *)) {
+    arc4random_buf(buf, nbytes);
+  }else {
+    OSStatus status = SecRandomCopyBytes(kSecRandomDefault, nbytes, buf);
+    if (status != errSecSuccess) {
+      fatalError(0, "Fatal error: SecRandomCopyBytes failed with error %d\n", status);
+    }
+  }
+}
+#elif defined(__linux__)
+#include <linux/version.h>
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(3,17,0)
+#include <sys/syscall.h>
+SWIFT_RUNTIME_STDLIB_INTERFACE
+void swift::_stdlib_random(void *buf, __swift_size_t nbytes) {
+  int result = syscall(SYS_getrandom, buf, nbytes, 0);
+  if (result != nbytes) {
+    fatalError(0, "Fatal error: Unexpected error with getrandom\n")
+  }
+}
+#else
+#include <mutex>
+static std::mutex m;
+static int fd;
+SWIFT_RUNTIME_STDLIB_INTERFACE
+void swift::_stdlib_random(void *buf, __swift_size_t nbytes) {
+  if (!fd) {
+    std::lock_guard<std::mutex> lock(m);
+    if (fd) {
+      swift::_stdlib_random(buf, nbytes);
+      return;
+    }
+    int oflags = O_RDONLY;
+    fd = swift::_stdlib_open("/dev/urandom", oflags);
+    if (fd < 0) {
+      fatalError(0, "Fatal error: Unable to open /dev/urandom\n");
+    }
+  }
+  if (swift::_stdlib_read(fd, buf, nbytes) < 0) {
```

Comment:
Reading from "/dev/urandom" (using `getrandom` or `_stdlib_read`)
can return fewer bytes than requested, so you might want to repeat
until the buffer is filled (or an error occurs).
<https://llvm.org/svn/llvm-project/libcxx/trunk/src/random.cpp>

Reply:
Thanks!